contract/src/actions.cpp

```diff
@@ -156,7 +156,7 @@ void check_result( ValidationResult r, const Transaction& txn, const char* desc
     }
 
     eosio::print( "ERR: ", uint64_t(r), "\n" );
-    eosio::check( false, desc );
+    eosio::check( false, std::string(desc)+ ": " + std::to_string(uint64_t(r)));
```

Comment:
There is a `check` that takes an assert code:
```
    /**
    *  Assert if the predicate fails and use the supplied error code.
    *
    *  @ingroup system
    *
    *  Example:
    *  @code
    *  eosio::check(a == b, 13);
    *  @endcode
    */
   inline void check(bool pred, uint64_t code) {
      if (!pred) {
         internal_use_do_not_use::eosio_assert_code(false, code);
      }
   }
```

Reply:
We agreed to go with the approach where we provide the meaningful error description directly from the contract instead.